tests/e2e/parallel/main_test.go

```diff
@@ -2248,6 +2248,498 @@ func TestScheduledSuiteUpdate(t *testing.T) {
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-invalid", testName),
+			Namespace: testNamespace,
+		},
+		Spec: compv1alpha1.CustomRuleSpec{
+			RulePayload: compv1alpha1.RulePayload{
+				ID:          fmt.Sprintf("custom_invalid_%s", testName),
+				Title:       "Invalid Rule",
+				Description: "This rule has invalid CEL expression",
+				Severity:    "low",
+			},
+			CustomRulePayload: compv1alpha1.CustomRulePayload{
+				ScannerType: compv1alpha1.ScannerTypeCEL,
+				Expression: `
+					pods.items.all(pod, 
+						invalid_function_that_doesnt_exist(pod)
+					)
+				`,
+				Inputs: []compv1alpha1.InputPayload{
+					{
+						Name: "pods",
+						KubernetesInputSpec: compv1alpha1.KubernetesInputSpec{
+							APIVersion: "v1",
+							Resource:   "pods",
+						},
+					},
+				},
+				ErrorMessage: "This should fail validation",
+			},
+		},
+	}
+
+	err := f.Client.Create(context.TODO(), invalidRule, nil)
+	if err != nil {
+		t.Fatalf("Failed to create CustomRule: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), invalidRule)
+
+	// Wait and expect the rule to have Error status
+	err = f.WaitForCustomRuleStatus(testNamespace, fmt.Sprintf("%s-invalid", testName), "Error")
+	if err != nil {
+		t.Fatalf("CustomRule validation failed: %v", err)
+	}
+	t.Log("CustomRule validation correctly rejected invalid expression")
+
+	// Test 2: Rule with undeclared variable
+	undeclaredVarRule := &compv1alpha1.CustomRule{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-undeclared", testName),
+			Namespace: testNamespace,
+		},
+		Spec: compv1alpha1.CustomRuleSpec{
+			RulePayload: compv1alpha1.RulePayload{
+				ID:          fmt.Sprintf("custom_undeclared_%s", testName),
+				Title:       "Undeclared Variable Rule",
+				Description: "This rule uses undeclared variables",
+				Severity:    "low",
+			},
+			CustomRulePayload: compv1alpha1.CustomRulePayload{
+				ScannerType: compv1alpha1.ScannerTypeCEL,
+				Expression: `
+					pods.items.all(pod,
+						deployments.items.exists(d, d.metadata.name == pod.metadata.name)
+					)
+				`,
+				Inputs: []compv1alpha1.InputPayload{
+					{
+						Name: "pods",
+						KubernetesInputSpec: compv1alpha1.KubernetesInputSpec{
+							APIVersion: "v1",
+							Resource:   "pods",
+						},
+					},
+					// 'deployments' is used but not declared as input
+				},
+				ErrorMessage: "This should fail validation due to undeclared variable",
+			},
+		},
+	}
+
+	err = f.Client.Create(context.TODO(), undeclaredVarRule, nil)
+	if err != nil {
+		t.Fatalf("Failed to create CustomRule: %v", err)
+	}
+	defer f.Client.Delete(context.TODO(), undeclaredVarRule)
+
+	// Wait and expect the rule to have Error status
+	err = f.WaitForCustomRuleStatus(testNamespace, fmt.Sprintf("%s-undeclared", testName), "Error")
+	if err != nil {
+		t.Fatalf("CustomRule validation failed: %v", err)
+	}
+	t.Log("CustomRule validation correctly detected undeclared variable")
```

Comment:
We could add a case here where we create a CEL rule that is valid, create a tailored profile with the valid rule, create an ssb with the tailored profile, then update the CEL rule expression to something invalid. At that point, we should be able to assert the rule is in error state, the profile is invalid, and the binding is invalid or not ready. Then fixing the expression should put everything back into a good state for all resources.

This would ensure we're testing the watchers across resources.